app/instrumentation-tests/src/org/commcare/androidTests/AppUpdateTest.kt

```diff
@@ -0,0 +1,195 @@
+import androidx.test.espresso.action.ViewActions.click
+import androidx.test.espresso.assertion.ViewAssertions.doesNotExist
+import androidx.test.espresso.assertion.ViewAssertions.matches
+import androidx.test.espresso.matcher.ViewMatchers.*
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.filters.LargeTest
+import androidx.test.platform.app.InstrumentationRegistry
+import org.commcare.dalvik.R
+import org.commcare.utils.InstrumentationUtility
+import org.hamcrest.Matchers.endsWith
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@RunWith(AndroidJUnit4::class)
+@LargeTest
+class AppUpdateTest: BaseTest() {
+
+    companion object {
+        const val CCZ_NAME = "app_update.ccz"
+        const val APP_NAME = "Update Test"
+    }
+
+    @Before
+    fun setup() {
+        installApp(APP_NAME, CCZ_NAME)
+        InstrumentationUtility.login("user_with_no_data", "123")
+    }
+
+    @Test
+    fun testAppUpdate() {
+        InstrumentationUtility.enableDeveloperMode()
+        // Enable app update item.
+        InstrumentationUtility.openOptionsMenu()
+        onView(withText("Settings"))
+                .perform(click())
+        onView(withText("Developer Options"))
+                .perform(click())
+        onView(withText("Show Update Options Item"))
+                .perform(click())
+        onView(withText("Enabled"))
+                .perform(click())
+        pressBack()
+
+        // Make sure the update endpoint is set to "Latest starred version"
+        onView(withText("Update Options"))
+                .perform(click())
+        onView(withText("Latest starred version"))
+                .perform(click())
+        pressBack()
+
+        // check base form content
+        onView(withText("Start"))
+                .perform(click())
+        onView(withText("Module Three"))
+                .check(matches(isDisplayed()))
+        onView(withText("Module One"))
+                .perform(click())
+        onView(withText("Example 1"))
+                .perform(click())
+        onView(withText("A text question"))
+                .check(matches(isDisplayed()))
+        closeSoftKeyboard()
+        pressBack()
+        onView(withText(R.string.do_not_save))
+                .perform(click())
+        pressBack()
+
+        // make sure case list doesn't have status column
+        onView(withText("Module Two"))
+                .perform(click())
+        onView(withText("Update Case"))
+                .perform(click())
+        onView(withText("Status"))
+                .check(doesNotExist())
+        pressBack()
+        pressBack()
+        pressBack()
+
+        // download the app update
+        InstrumentationUtility.openOptionsMenu()
+        onView(withText("Update App"))
+                .perform(click())
+        onView(withText("Current version: 2"))
+                .check(matches(isDisplayed()))
+        onView(withText("Update to version 11 & log out"))
```

Comment:
Ahh okay. One of the reasons I didn't do it is because espresso doesn't support rotating devices. I'll use `uiautomator` to rotate the device.

Reply:
@shubham1g5 I came back to this today, and I guess you're hinting at stopping the update in between. 
And I don't think it's possible to click on the stop button(since espresso is gonna wait for the update to finish before doing any operation) 😅  
I'll try to do it using internal commcare methods maybe, but yeah just an FYI.

Reply:
Interestingly, I never removed this stuff 🤣 https://github.com/dimagi/commcare-android/blob/ce581c30ff966e0676db66092822e3283583d7f3/app/instrumentation-tests/AndroidManifest.xml#L6